src/components/FileTreeNode/index.tsx

```diff
@@ -73,4 +85,15 @@ class FileTreeNodeBase<TreeNodeType> extends React.Component<PublicProps> {
   }
 }
 
-export default FileTreeNodeBase;
+const mapStateToProps = (state: ApplicationState, ownProps: PublicProps) => {
+  const { versionId } = ownProps;
+
+  const version = getVersionInfo(state.versions, versionId);
+  const selectedPath = version ? version.selectedPath : undefined;
```

Comment:
If `version` is not loaded, `selectedPath` would be undefined and the component cannot repair itself because it does not know how to fetch versions. When I see this pattern, it tells me that the component should not be mapping its props from Redux state. 

I suggest accepting `version` or `selectedPath` as a public prop instead. This would put the burden of configuration on the code that calls `FileTreeNode` but that seems fine to me. It's already configuring it with `versionId` anyway. Switching to that approach will address the problem up above and will make testing easier.